transactions/src/main/java/org/jboss/as/txn/subsystem/TransactionSubsystemRootResourceDefinition.java

```diff
@@ -245,6 +252,7 @@ public class TransactionSubsystemRootResourceDefinition extends SimpleResourceDe
     public static final SimpleAttributeDefinition JDBC_STORE_DATASOURCE = new SimpleAttributeDefinitionBuilder(CommonAttributes.JDBC_STORE_DATASOURCE, ModelType.STRING, true)
             .setFlags(AttributeAccess.Flag.RESTART_JVM)
             .setXmlName(Attribute.DATASOURCE_JNDI_NAME.getLocalName())
+            .setCapabilityReference(DATA_SOURCE_CAPABILITY_NAME, TRANSACTION_CAPABILITY)
```

Comment:
This does not look correct to me.
According to:
https://github.com/wildfly/wildfly/blob/main/transactions/src/main/java/org/jboss/as/txn/subsystem/TransactionSubsystemAdd.java#L347
... and ...
https://github.com/wildfly/wildfly/blob/main/transactions/src/main/java/org/jboss/as/txn/subsystem/TransactionSubsystemAdd.java#L371
... the JDBC_STORE_DATASOURCE attribute expects a JNDI name, not a reference to a DataSource resource.

To implement this correctly, you will need to add a new attribute (as an alternative to the existing one) that accepts a data-source reference, and deprecate the existing attribute.  The resource add operation handler would then resolve the DataSource dependency via its capability reference, if defined.

Reply:
@pferraro I see, I would've expected some validation in that case. 

That said the "capability reference" doesn't change the datatype, we're still passing a string, in this case the basic name of the datasource. Unfortunately I'm not seeing a way to obtain the JNDI name from the capability reference so I might have to scratch this.

Reply:
It is not possible to configure the Arjuna ObjectStore with a DataSource instance?

Reply:
I'm not familiar enough with it to say, but the current implementation uses some string-based config - https://github.com/wildfly/wildfly/blob/9d48cbde13d7da3db315161ccb5aa1cfd2308b50/transactions/src/main/java/org/jboss/as/txn/service/ArjunaObjectStoreEnvironmentService.java#L107-L109

I will take this to Zulip, I can't get this to work even with the proper configuration